### PR TITLE
Fix broken links in editor documentation

### PR DESCRIPTION
**Pull request by derekblank**
## What?
Fixes a handful of broken links in the Editor documentation.

## Why?
To encourage external code contributions, and also not have broken links in documentation.

## How?
Finds and replaces the related updated link URL for broken links.

## Testing Instructions
Referenced documentation links should not 404.

### CODE REVIEW

File: packages/react-native-editor/__device-tests__/CONTRIBUTING.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ Using one or a combination of these tools will make it much easier to identify w
 
 -   You'll write any functions needed to interact with the page in the `EditorPage` page object and then call those interactions within the test. The code you'll need to write to actually do the finding will use a combination of
 
--   Appium's spec https://appium.io/docs/en/about-appium/intro/ which you can find examples of a variety of functions under the commands tab
```
Comment (fluiddot) [inline packages/react-native-editor/__device-tests__/CONTRIBUTING.md:41]: We don't use Appium 2 yet ([reference](https://github.com/WordPress/gutenberg/blob/bdf728b7bf7105ad7ee576200aa46dc82ff424fb/package.json#LL166C4-L166C8)) so not sure if we should point to this documentation. I haven't found the docs for version 1.x, which makes me think that might have been removed, as version 1.x is no longer maintained 🤔. If so, we could add a note about this, WDYT?

Comment (derekblank) [reply packages/react-native-editor/__device-tests__/CONTRIBUTING.md:41]: That's a good point -- Appium's docs have moved around quite a bit recently with the 2.0 release. I couldn't find the 1.x docs initially, but eventually found them on [GitHub](https://github.com/appium/appium/tree/1.x/docs/en), and updated the link references. 👍

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR, @derekblank.

While reviewing this PR, I found another broken link regarding Appium:

https://github.com/WordPress/gutenberg/blob/5a1d3760bef4abc7758961d4d17bf05f7fee90da/docs/contributors/code/react-native/getting-started-react-native.md?plain=1#L114

Instead of https://appium.io/docs/en/about-appium/getting-started/, https://github.com/appium/appium/blob/1.x/docs/en/about-appium/getting-started.md may be the currently valid link.

If you would like, how about responding with this PR?

**Comment by derekblank** [comment]
@t-hamano Thanks for reviewing, and great suggestion! I agree that linking to GitHub would likely be more stable. I've updated the docs. 👍

**Review by t-hamano — APPROVED** [review]
LGTM! Thanks for the update 👍

**Pull request merged**